Makefile

```
@@ -44,11 +44,13 @@ $(GENERATORBIN):
 $(GENERATORBIN)/generator: $(G_SRCS) | $(GENERATORBIN)
 	@crystal build $(GENERATORDIR)/generator.$(FILEEXT) -o generator/bin/generator
 
-build-generator: $(GENERATORBIN)/generator
+build-generator:
+	@echo "building binary for generator: $(GENERATOR)"
+	@cd $(GENERATORDIR) && (shards check || shards install) && mkdir -p bin && crystal build generator.cr -o bin/generator
 
-generate-exercise: $(GENERATORBIN)/generator
+generate-exercise:
```

```suggestion
.PHONY: generate-exercise
generate-exercise:
```

(Same reasoning as `build-generator`)